### PR TITLE
login is aligned with nav links when language & country unselected

### LINKED ISSUE #2671: Mobile drawer UI fix
We would like to realign the content inside the Header and Drawer like thoses designs in [Figma](https://www.figma.com/file/LlopBqbeLND3IQQvYMjVbn/%F0%9F%9A%A7-Navigation-UX-%E2%80%93-Dawn-2023?type=design&node-id=2712%3A109199&t=ERNnDIdYRVGM6QG2-1).

- [ ] Burger/Close button
- [ ] Icons group (apps/cart/search)
- [ ] Menu items/Arrows
- [ ] Language
- [ ] Country/currency
- [ ]  Social icons
- [ ] Account icon/label

On mobile only.

### PR DESCRIPTION
**Pull request by lougoncharenko**
### PR Summary: 
When mobile drawer is login only, the login link is aligned with the nav links.

### Why are these changes introduced?

Fixes #2671 .

### What approach did you take?
I used a css class to apply the margin-left when the utility links were missing the localization form but had the account.

### Other considerations
N/A

### Visual impact on existing themes
If merchants choose to unselect the country and language selector, the account login button will be aligned on the mobile version.

### Testing steps/scenarios

- [ ] View Mobile drawer with country and language selectors checked
- [ ] View Mobile drawer with country selector checked
- [ ] View Mobile drawer with language selector checked
- [ ] View Mobile Drawer with both country and language selectors unchecked (account only)
- [ ] View Mobile Drawer with the social icons and account only

### Demo links
- [Editor](https://os2-demo.myshopify.com/admin/themes/140174163990/editor?context=theme&category=gid%3A%2F%2Fshopify%2FOnlineStoreThemeSettingsCategory%2FSocial%2Bmedia%3Ftheme_id%3D140174163990%26first_setting_id%3Dsocial_facebook_link)

### Checklist
- [ ] Added PR summary for [release notes](https://themes.shopify.com/themes/dawn/styles/default#ReleaseNotes)
- [ ] Requested review from UX (Only for changes that are affecting the experience or perceivable visual details)
- [ ] Created a ticket for the [help.shopify.com](https://help.shopify.com) documentation team about updates to theme settings. (Internal-only task)
- [ ] Followed [theme code principles](https://github.com/Shopify/dawn/blob/main/.github/CONTRIBUTING.md#theme-code-principles)
- [ ] Linted with [Theme Check](https://github.com/Shopify/theme-check)
- [ ] Tested on [mobile](https://shopify.dev/themes/store/requirements#mobile-browser-requirements)
- [ ] Tested on [multiple browsers](https://shopify.dev/themes/store/requirements#desktop-browser-requirements)
- [ ] Tested for [accessibility](https://shopify.dev/themes/best-practices/accessibility)

### CODE REVIEW

File: assets/component-menu-drawer.css [hunk 1]
```diff
@@ -237,7 +237,14 @@ details[open].menu-opening > .menu-drawer__submenu {
   margin-bottom: 0;
 }
 
-.menu-drawer__utility-links:has(ul:empty) .menu-drawer__account,
+.menu-drawer__utility-links:where(:not(.menu-drawer__localization)) .menu-drawer__account {
+  margin-left: 1rem;
+}
+
+.header--has-social .menu-drawer__utility-links:where(:not(.menu-drawer__localization)) .menu-drawer__account {
+  margin-left: 0rem;
```
Comment (kmeleta) [inline assets/component-menu-drawer.css:245]: nitpick, `0rem` -> `0`

**Review by kmeleta — COMMENTED** [review]
LGTM! Just a couple small code suggestions.

**Review by kmeleta — APPROVED** [review]

**Comment by kimberlyoleiro** [comment]
Hey! *Nit-pick

Curious why the padding is 30px when lang/currency is present, but 20px + 10px margin without it?
Group 5

File: assets/component-menu-drawer.css [hunk 2]
```diff
@@ -217,7 +217,7 @@ details[open].menu-opening > .menu-drawer__submenu {
 
 @media screen and (max-width: 749px) {
   .header--has-account:where(:not(.header--has-social):not(.header--has-localizations)) .menu-drawer__utility-links {
-    padding: 2rem;
+    padding: 3rem;
```
Comment (ludoboludo) [inline assets/component-menu-drawer.css:220]: In other scenarios we're using `2rem 3rem`. Maybe we could do the same here as I don't think it needs the extra `1rem` of padding at the top/bottom.

Comment (ludoboludo) [reply assets/component-menu-drawer.css:220]: Same below on line `226`. Maybe we can confirm with a designer wether we should apply a consistent `2rem 3rem` padding or sometimes have `3rem` and sometimes not 🤔

Comment (lougoncharenko) [reply assets/component-menu-drawer.css:220]: these are really good suggestions! And a good point to bring up! I am updating the code to keep it consistent

**Review by kmeleta — APPROVED** [review]

**Pull request merged**